`numpy>=2.0` failures -- [2024-06-19]

`numpy>=2.0` failures to fix before #455 

* https://github.com/pysal/spopt/actions/runs/9574404779/job/26397585653

```
FAILED spopt/tests/test_maxp.py::TestMaxPHeuristic::test_maxp_heuristic_complex - AttributeError: `np.Inf` was removed in the NumPy 2.0 release. Use `np.inf` instead.. Did you mean: 'inf'?
FAILED spopt/tests/test_maxp.py::TestMaxPHeuristic::test_maxp_one_var - AttributeError: `np.Inf` was removed in the NumPy 2.0 release. Use `np.inf` instead.. Did you mean: 'inf'?
```

--- a/spopt/region/maxp.py
+++ b/spopt/region/maxp.py
@@ -562,7 +562,7 @@ def _check_move(
     lost_distance = distance_matrix[poa, rm].sum()
     potential_move = None
 
-    min_added_distance = np.Inf
+    min_added_distance = np.inf
     for poan in poa_neighbor:
         recipient_region = labels[poan]
         if donor_region != recipient_region:

